xarray/tests/test_backends.py

```diff
@@ -881,6 +883,41 @@ class NetCDF4ViaDaskDataTestAutocloseTrue(NetCDF4ViaDaskDataTest):
     autoclose = True
 
 
+@requires_zarr
+class ZarrDataTest(CFEncodedDataTest, TestCase):
+    @contextlib.contextmanager
+    def create_store(self):
+        with create_tmp_file(suffix='.zarr') as tmp:
+            yield backends.ZarrStore(store=tmp)
+
+    @contextlib.contextmanager
+    def roundtrip(self, data, save_kwargs={}, open_kwargs={},
+                  allow_cleanup_failure=False):
+        with create_tmp_file(suffix='.zarr',
+                allow_cleanup_failure=allow_cleanup_failure) as tmp_file:
+            data.to_zarr(store=tmp_file, **save_kwargs)
+            yield xr.open_zarr(tmp_file, **open_kwargs)
+
+    def test_auto_chunk(self):
+        original = create_test_data ().chunk()
+
+        with self.roundtrip(original,
+                open_kwargs={'auto_chunk': False}) as actual:
+            for k, v in actual.variables.items():
+                # only index variables should be in memory
+                self.assertEqual(v._in_memory, k in actual.dims)
+                # there should be no chunks
+                self.assertEqual(v.chunks, None)
+
+        with self.roundtrip(original,
+                open_kwargs={'auto_chunk': True}) as actual:
+            for k, v in actual.variables.items():
+                # only index variables should be in memory
+                self.assertEqual(v._in_memory, k in actual.dims)
+                # chunk size should be the same as original
+                self.assertEqual(v.chunks, original[k].chunks)
```

Comment:
I do not understand why my `auto_chunk` option is not working. The variables should be dask arrays here, but they are not...that's why the test is failing.

Reply:
I fixed this by moving the auto_chunking logic out of `ZarrStore` and into `open_zarr`. The reason is that the variable chunks set by the store did not survive `decode_cf_variable`.

Reply:
Oof... We desperately need to improve this!